flow/project.py

```diff
@@ -687,11 +660,37 @@ def _determine_entrypoint(self, entrypoint, directives, job):
         else:
             return "{} {}".format(entrypoint['executable'], entrypoint['path']).lstrip()
 
-    def _resolve_directives(self, name, defaults):
+    def _resolve_directives(self, name, defaults, job):
         if name in self.operation_directives:
-            return deepcopy(self.operation_directives[name])
+            directives = deepcopy(self.operation_directives[name])
         else:
-            return deepcopy(defaults.get(name, dict()))
+            directives = defaults.get(name, dict())
```

Comment:
We want a deepcopy here to ensure that we don't write over the default dictionary provided.